onedal/basic_statistics/tests/test_basic_statistics.py

```diff
@@ -98,3 +106,79 @@ def test_option_weighted(queue, option, dtype):
 
         tol = fp32tol if res.dtype == np.float32 else fp64tol
         assert_allclose(gtr, res, rtol=tol)
+
+    @pytest.mark.parametrize("queue", get_queues())
+    @pytest.mark.parametrize("dtype", [np.float32, np.float64])
+    def test_basic_csr(queue, dtype):
+        from packaging.version import Version
+        from scipy import __version__
+
+        # Support of scipy.sparse.random_array was added in scipy 1.12.0
+        # Deselecting the test for earlier versions
+        if Version(__version__) < Version("1.12.0"):
+            pytest.skip("skipping BasicStatistics CSR test for SciPy < 1.12.0")
+
+        seed = 42
+        s_count, f_count = 5000, 3008
+
+        gen = np.random.default_rng(seed)
+
+        from scipy.sparse import random_array
+
+        data = random_array(
+            shape=(s_count, f_count),
+            density=0.01,
+            format="csr",
+            dtype=dtype,
+            random_state=gen,
+        )
+
+        alg = BasicStatistics(result_options="mean")
+        res = alg.compute(data, queue=queue)
+
+        res_mean = res["mean"]
+        gtr_mean = data.mean(axis=0)
+        tol = 2e-5 if res_mean.dtype == np.float32 else 1e-7
+        assert_allclose(gtr_mean, res_mean, rtol=tol)
+
+    @pytest.mark.parametrize("queue", get_queues())
+    @pytest.mark.parametrize("option", options_and_tests_csr)
+    @pytest.mark.parametrize("dtype", [np.float32, np.float64])
+    def test_options_csr(queue, option, dtype):
+        from packaging.version import Version
+        from scipy import __version__
+
+        # Support of scipy.sparse.random_array was added in scipy 1.12.0
+        # Deselecting the test for earlier versions
+        if Version(__version__) < Version("1.12.0"):
+            pytest.skip("skipping BasicStatistics CSR test for SciPy < 1.12.0")
```

Comment:
please avoid code duplication, check this globally use for test suits

Reply:
Done